frontend/src/context/UserContext.tsx

```diff
@@ -54,7 +55,7 @@ export const UserProvider = ({ children }: { children: React.ReactNode }) => {
     };
 
     fetchData();
-  }, [web3]);
+  }, [web3, isAccountChanged]);
```

Comment:
@ZYJLiu what happens here if you switch to an account that you've never approved to connect before? I imagine we'd have to go through the login process again there

Reply:
If a user changes their connected metamask account, you can just update the state where you're storing the user's address to reflect that. In our demo app we just do `setUser(newAccount)`